src/java/org/apache/cassandra/index/sai/disk/v7/V7OnDiskFormat.java

```diff
@@ -32,19 +31,63 @@ public class V7OnDiskFormat extends V6OnDiskFormat
+                                                                                IndexComponentType.KD_TREE,
+                                                                                IndexComponentType.KD_TREE_COMPRESSION_INFO,
+                                                                                IndexComponentType.KD_TREE_POSTING_LISTS,
+                                                                                IndexComponentType.KD_TREE_POSTING_LISTS_COMPRESSION_INFO);
+
+
     private static final Set<IndexComponentType> LITERAL_COMPONENTS = EnumSet.of(IndexComponentType.COLUMN_COMPLETION_MARKER,
                                                                                  IndexComponentType.META,
                                                                                  IndexComponentType.TERMS_DATA,
+                                                                                 IndexComponentType.TERMS_DATA_COMPRESSION_INFO,
                                                                                  IndexComponentType.POSTING_LISTS,
+                                                                                 IndexComponentType.POSTING_LISTS_COMPRESSION_INFO,
                                                                                  IndexComponentType.DOC_LENGTHS);
 
+    private static final Set<IndexComponentType> COMPRESSION_INFO_COMPONENTS = EnumSet.of(IndexComponentType.POSTING_LISTS_COMPRESSION_INFO,
+                                                                                          IndexComponentType.TERMS_DATA_COMPRESSION_INFO,
+                                                                                          IndexComponentType.KD_TREE_COMPRESSION_INFO,
+                                                                                          IndexComponentType.KD_TREE_POSTING_LISTS_COMPRESSION_INFO,
+                                                                                          IndexComponentType.PRIMARY_KEY_TRIE_COMPRESSION_INFO,
+                                                                                          IndexComponentType.PRIMARY_KEY_BLOCKS_COMPRESSION_INFO);
+
+
     @Override
     public Set<IndexComponentType> perIndexComponentTypes(AbstractType<?> validator)
     {
         if (validator.isVector())
             return V3OnDiskFormat.VECTOR_COMPONENTS_V3;
         if (TypeUtil.isLiteral(validator))
             return LITERAL_COMPONENTS;
-        return V1OnDiskFormat.NUMERIC_COMPONENTS;
+        return NUMERIC_COMPONENTS;
+    }
+
+    @Override
+    public Set<IndexComponentType> perSSTableComponentTypes()
+    {
+        return PER_SSTABLE_COMPONENTS;
+    }
+
+    /**
+     * Returns the components storing compression info about compression of other index components.
+     * Compression information components are not present on disk if the index component they describe is not compressed.
+     */
```

Comment:
Nit: this doc might be better placed on the interface method, `OnDiskFormat#compressionInfoComponentTypes`.